packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -79,8 +79,26 @@ const getAccountInfo = async (request: Request<MessageTypes.GetAccountInfo>) =>
 
     const transactionsPage = await fetchTransactionPage(request);
 
+    const uniqueTransactionSlots = Array.from(new Set(transactionsPage.map(tx => tx.slot)));
+
+    const slotToBlockHeightMapping = (
```

Comment:
We do not get blockheight from the transaction history, we only get slot for each transaction. For this reason we have to fetch block for each slot and an create a dictionary

Let me point out that this is the slowest part, getting the mapping takes 90% of the time of getting the tx page, which is quite sad. We should test this with a higher RPC node plan to see if we need to do something about it, for now I haven't figured out a way round it :/